iceoryx_posh/include/iceoryx_posh/popo/publisher.hpp

```diff
@@ -45,16 +46,8 @@ class PublisherInterface
 template <typename T, typename H = mepoo::NoUserHeader, typename BasePublisher_t = BasePublisher<>>
 class PublisherImpl : public BasePublisher_t, private PublisherInterface<T, H>
 {
-    static_assert(!std::is_void<T>::value, "The type `T` must not be void. Use the UntypedPublisher for void types.");
-    static_assert(!std::is_void<H>::value, "The user-header `H` must not be void.");
-
-    static_assert(!std::is_const<T>::value, "The type `T` must not be const.");
-    static_assert(!std::is_reference<T>::value, "The type `T` must not be a reference.");
-    static_assert(!std::is_pointer<T>::value, "The type `T` must not be a pointer.");
-
-    static_assert(!std::is_const<H>::value, "The user-header `H` must not be const.");
-    static_assert(!std::is_reference<H>::value, "The user-header `H` must not be a reference.");
-    static_assert(!std::is_pointer<H>::value, "The user-header must `H` not be a pointer.");
+    using DataTypeAssert = typename TypedPortApiTrait<T>::Assert;
```

Comment:
This works too to avoid template args (and we keep the fine grained errors of the static asserts).